instrumentation/activej-http-6.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/activejhttp/ActivejAsyncServletInstrumentation.java

```diff
@@ -52,39 +54,48 @@ public void transform(TypeTransformer transformer) {
   @SuppressWarnings("unused")
   public static class ServeAdvice {
 
+    public static class AdviceLocals {
```

Comment:
A while ago we already attempted some conversions. In one of these we used https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/netty/netty-common-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v4/common/NettyScope.java Perhaps we should also use a similar pattern here. Instead of having a class that has only the fields we wish to pass from start to end we could move some logic into it. While `NettyScope` has a static end method we could also look into having an instance method for this so that access to the locals wouldn't need to go through a variable. WDYT?

Reply:
I just updated the PR after migrating those "advice local" classes to "advice scope" and migrating  the code from the advice to `AdviceScope`, this requires a bit of manual work but seems quite close to what we previously did with `HibernateOperationScope` or `NettyScope`, let me know if that matches what you had in mind here.